src/View/Helper/LicenseHelper.php

```diff
@@ -20,75 +20,76 @@
 
 use App\View\Helper\AppHelper;
 use Cake\View\View;
-
+use App\Lib\Licenses;
 
 class LicenseHelper extends AppHelper
 {
     public $helpers = array(
         'Html',
     );
 
-    private $licenses;
+    /* Contains data for all licenses used on Tatoeba */
+    protected $licenses = [];
+
+    /* The licenses used in the current context (sentences or audio files) */
+    protected $validLicenses = [];
 
-    public function __construct(View $view, $config = array()) {
-        assert(isset($config['availableLicences']));
-        parent::__construct($view, $config);
-        $this->licenses = array(
-            '' => array('name' => __('No license for offsite use')),
```

Comment:
The audio license string `''` and its name "No license for offsite use" need to be kept because they are currently used for audio recordings which have been contributed without an explicit license. When this is the case, the audio page of a user looks like that for example: https://tatoeba.org/eng/audio/of/dcampbell

A bit of history. In the past, there were no "Audio" block on the sentence page, and no `audios` table. Just a `has_audio` column in the `sentences` table and links to mp3 files on the speaker icon. So a lot of audio have been initially added at a time where we didn’t really think about licensing. Only later I changed it the way it is now, in order give audio contributors more recognition for their work. Still now, it looks like some audio contributors are not choosing a license. I believe it’s because they need to choose a license by themselves from that page, and a lot of them don’t or don’t know or don’t care about the license.

Reply:
I've written that version under the premise that we come up with a good name for the empty license string for both sentences and audio files as [discussed in the issue](https://github.com/Tatoeba/tatoeba2/issues/1659#issuecomment-606251828). But it looks like it is probably better to don't mix them.

Since we use completely different licenses for sentences and audio files, my new version will use two different arrays for them.

> I believe it’s because they need to choose a license by themselves from that page, and a lot of them don’t or don’t know or don’t care about the license.

I've noticed that with the current version from the dev branch I can't choose the option "No license for offsite use" in the selection because the FormHelper adds the `required` attribute to the `select` element which doesn't allow to choose an option with the empty string value. Can you confirm this? Is that on purpose?

Reply:
No it’s not on purpose. I confirm there is a problem with license selection box.

Reply:
I've found out why the `required` attribute is added. The FormHelper uses the table validation rules (when in entity context) in order to decide whether to add it or not:
https://github.com/Tatoeba/tatoeba2/blob/24fb3bb9206e04e6899a41a247b51d155cfb3de7/src/Model/Table/UsersTable.php#L139-L141
After adding `->allowEmptyString('audio_license')` there's no `required` attribute any more.

So should all users be able to choose `No license for offsite use` or just admins?

Reply:
All users can choose `No license for offsite use` for their audio contribution, that’s the current behaviour.